Stellar-transaction.x

```diff
@@ -530,12 +530,31 @@ case HOST_FUNCTION_TYPE_INSTALL_CONTRACT_CODE:
     InstallContractCodeArgs installContractCodeArgs;
 };
 
+struct AuthorizedInvocation
+{
+    Hash contractID;
+    SCSymbol functionName;
+    SCVec args;
+    AuthorizedInvocation subInvocations<>;
+};
+
+struct ContractAuth
+{
+    SCAddress* address; // not present for invoker
+    AuthorizedInvocation invocation;
+    SCVec signatureArgs;
+    uint64* nonce; // not present for invoker
```

Comment:
`ContractAuth` might be more extensible, and clearer, if we make it a union where Invoker only has two fields, and other auth forms have all four. Or a struct containing a union where the two fields are in the struct plus a union with additional fields.

Reply:
I'm just still not quite sure about special-casing the invoker at all. It actually could be safer to always explicitly specify the address and maybe even use nonce (so only skipping the signature is allowed). But if we stick to special-casing invoker, I agree that maybe we should group these together.

Reply:
That's a good point 👍🏻.

There may be value in introducing a union here somewhere so that we can take a second stab at auth if we need to. Although, I guess we can add a new operation for that if we need to. So yeah, this is fine.

Reply:
Union here would definitely be an easier upgrade than `InvokeHostFunctionV2` though.